provider/source/src/cldr_serde/units/info.rs

```diff
@@ -80,3 +81,54 @@ pub(crate) struct Supplemental {
 pub(crate) struct Resource {
     pub(crate) supplemental: Supplemental,
 }
+
+impl Resource {
+    /// Retrieves the unique identifier for a given unit name.
+    ///
+    /// This function searches for the specified unit name within the `convert_units`
+    /// and returns its position as a `u16` identifier.
+    ///
+    /// # Arguments
+    ///
+    /// * `unit_name` - A string slice that holds the name of the unit to search for.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(u16)` - The unique identifier for the unit if found.
+    /// * `Err(DataError)` - An error if the unit is not found or if the index is out of range for `u16`.
```

Comment:
observation: this is very verbose for an internal function

Reply:
done